common/rpc/factory.go

```diff
@@ -126,35 +140,88 @@ func NewFactory(logger log.Logger, p Params) *Factory {
 	dispatcher := yarpc.NewDispatcher(yarpc.Config{
 		Name:               p.ServiceName,
 		Inbounds:           inbounds,
-		Outbounds:          outbounds,
+		Outbounds:          outbounds.Outbounds,
 		InboundMiddleware:  p.InboundMiddleware,
 		OutboundMiddleware: p.OutboundMiddleware,
 	})
 
-	return &Factory{
+	ctx, cancel := context.WithCancel(context.Background())
+	return &FactoryImpl{
 		maxMessageSize: p.GRPCMaxMsgSize,
 		dispatcher:     dispatcher,
 		channel:        ch.Channel(),
+		outbounds:      outbounds,
+		serviceName:    p.ServiceName,
+		logger:         logger,
+		ctx:            ctx,
+		cancelFn:       cancel,
 	}
 }
 
 // GetDispatcher return a cached dispatcher
-func (d *Factory) GetDispatcher() *yarpc.Dispatcher {
+func (d *FactoryImpl) GetDispatcher() *yarpc.Dispatcher {
 	return d.dispatcher
 }
 
 // GetChannel returns Tchannel Channel used by Ringpop
-func (d *Factory) GetChannel() tchannel.Channel {
+func (d *FactoryImpl) GetTChannel() tchannel.Channel {
 	return d.channel
 }
 
-func (d *Factory) GetMaxMessageSize() int {
+func (d *FactoryImpl) GetMaxMessageSize() int {
 	if d.maxMessageSize == 0 {
 		return defaultGRPCSizeLimit
 	}
 	return d.maxMessageSize
 }
 
+func (d *FactoryImpl) Start(peerLister PeerLister) error {
+	// subscribe to membership changes and notify outbounds builder for peer updates
+	d.peerLister = peerLister
+	ch := make(chan *membership.ChangedEvent, 1000)
+	if err := d.peerLister.Subscribe(d.serviceName, factoryComponentName, ch); err != nil {
```

Comment:
fyi I think is is a method call now? you may need to merge in master

Reply:
`d.peerLister` is the membership resolver instance. Its interface didn't change.